ui/projects/streampipes/platform-services/src/lib/apis/datalake-rest.service.ts

```diff
@@ -19,190 +19,243 @@
-                       queryParams: any) {
-    const url = this.dataLakeUrl + '/measurements/' + index + '/download';
-    const request = new HttpRequest('GET', url, {
-      reportProgress: true,
-      responseType: 'text',
-      params: this.toHttpParams(queryParams)
-    });
-
-    return this.http.request(request);
-  }
 
-  toHttpParams(queryParamObject: any): HttpParams {
-    return new HttpParams({fromObject: queryParamObject});
-  }
-
-  removeData(index: string) {
-    const url = this.dataLakeUrl + '/measurements/' + index;
+    public get dataLakeUrl() {
+        return this.baseUrl + '/api/v4' + '/datalake';
+    }
 
-    return this.http.delete(url);
-  }
+    public get dataLakeMeasureUrl() {
+        return this.baseUrl + '/api/v4/datalake/measure';
+    }
 
-  dropSingleMeasurementSeries(index: string) {
-    const url = this.dataLakeUrl + '/measurements/' + index + '/drop';
-    return this.http.delete(url);
-  }
+    getAllMeasurementSeries(): Observable<DataLakeMeasure[]> {
+        const url = this.dataLakeUrl + '/measurements/';
+        return this.http.get(url).pipe(
+            map(response => {
+                return (response as any[]).map(p =>
+                    DataLakeMeasure.fromData(p),
+                );
+            }),
+        );
+    }
 
-  dropAllMeasurementSeries() {
-    const url = this.dataLakeUrl + '/measurements/';
-    return this.http.delete(url);
-  }
+    getMeasurement(id: string): Observable<DataLakeMeasure> {
+        return this.http
+            .get(`${this.dataLakeMeasureUrl}/${id}`)
+            .pipe(map(res => res as DataLakeMeasure));
+    }
 
-  private getQueryParameters(columns?: string, startDate?: number, endDate?: number, page?: number, limit?: number,
-                             offset?: number, groupBy?: string, order?: string, aggregationFunction?: string, timeInterval?: string):
-    DatalakeQueryParameters {
-    const queryParams: DatalakeQueryParameters = new DatalakeQueryParameters();
+    performMultiQuery(
+        queryParams: DatalakeQueryParameters[],
+    ): Observable<SpQueryResult[]> {
+        return this.http
+            .post(`${this.dataLakeUrl}/query`, queryParams, {
+                headers: { ignoreLoadingBar: '' },
+            })
+            .pipe(map(response => response as SpQueryResult[]));
+    }
 
-    if (columns) {
-      queryParams.columns = columns;
+    getData(
+        index: string,
+        queryParams: DatalakeQueryParameters,
+        ignoreLoadingBar?: boolean,
+    ): Observable<SpQueryResult> {
+        const columns = queryParams.columns;
+        if (columns === '') {
+            const emptyQueryResult = new SpQueryResult();
+            emptyQueryResult.total = 0;
+            return of(emptyQueryResult);
+        } else {
+            const url = this.dataLakeUrl + '/measurements/' + index;
+            const headers = ignoreLoadingBar ? { ignoreLoadingBar: '' } : {};
+            // @ts-ignore
```

Comment:
I think this is the reason why the PR validation failed.
Can you try to replace it with:
```
 return this.http.get<SpQueryResult>(url, {
   params: queryParams as unknown as HttpParams,
   headers,
});
```